# MANOVA 

Manova is a multivariate extension of ANOVA. It is used to model two or more dependent variables that are continuous with one or more categorical predictor variables. It is often used to assess for differences between two or more groups.

To perform a Multivariate Analysis of Variance (MANOVA) in Python, we typically use the `statsmodels` library. MANOVA is used when there are two or more dependent variables and one or more independent variables. It tests whether the mean differences among groups on a combination of dependent variables are likely to have occurred by chance.

Here's an example demonstrating how to create a MANOVA table in Python:

### Example: MANOVA with StatsModels

Let's say we have a dataset with two dependent variables (e.g., test scores in mathematics and science) and one independent variable (e.g., teaching method). We want to know if there are statistically significant differences in the dependent variables across the levels of the independent variable.

### Explanation:

- **Dataset Preparation**: The `data` dictionary and `DataFrame` (`df`) contain the sample data. Replace this with your actual data.
- **MANOVA Execution**: The `MANOVA.from_formula` method is used to perform the MANOVA. The formula 'MathScore + ScienceScore ~ Method' indicates that `MathScore` and `ScienceScore` are dependent variables, and `Method` is the independent variable.
- **Results**: The `mv_test()` method is used to get the MANOVA test results, which are printed to the console.

This script will output the MANOVA table, including Pillai's trace, Wilks' lambda, Hotelling-Lawley trace, and Roy's greatest root test statistics, along with their associated F-values, degrees of freedom, and p-values. These results will help you determine if there are statistically significant differences in the dependent variables across the levels of the independent variable.

In [1]:
import pandas as pd
import statsmodels.api as sm
from statsmodels.multivariate.manova import MANOVA

# Example dataset
data = {
    'Method': ['A', 'A', 'A', 'B', 'B', 'B', 'C', 'C', 'C'],
    'MathScore': [20, 22, 21, 19, 18, 20, 22, 23, 21],
    'ScienceScore': [30, 28, 29, 33, 32, 31, 29, 27, 28]
}

df = pd.DataFrame(data)
print(df.head(10))
# Perform MANOVA
maov = MANOVA.from_formula('MathScore + ScienceScore ~ Method', data=df)
print(maov.mv_test())

  Method  MathScore  ScienceScore
0      A         20            30
1      A         22            28
2      A         21            29
3      B         19            33
4      B         18            32
5      B         20            31
6      C         22            29
7      C         23            27
8      C         21            28
                   Multivariate linear model
                                                                
----------------------------------------------------------------
       Intercept          Value   Num DF Den DF  F Value  Pr > F
----------------------------------------------------------------
          Wilks' lambda    0.0005 2.0000 5.0000 4711.5000 0.0000
         Pillai's trace    0.9995 2.0000 5.0000 4711.5000 0.0000
 Hotelling-Lawley trace 1884.6000 2.0000 5.0000 4711.5000 0.0000
    Roy's greatest root 1884.6000 2.0000 5.0000 4711.5000 0.0000
----------------------------------------------------------------
                              

## **Interpertation of MANOVA Results**
The MANOVA results provided contain two main parts: the test statistics associated with the intercept and the test statistics associated with the independent variable (`Method`). Each part includes four different test statistics: Wilks' lambda, Pillai's trace, Hotelling-Lawley trace, and Roy's greatest root. Let's interpret these results:

### Intercept Part

1. **Wilks' Lambda**: A value close to 0 (0.0005) with a significant F-value (4711.5) and a p-value of 0.0000 indicates that the model with the intercept is significantly different from a model without the intercept.
2. **Pillai's Trace**: Similar to Wilks' lambda, a value close to 1 (0.9995) with a significant F-value and p-value indicates strong model significance.
3. **Hotelling-Lawley Trace**: A very high value (1884.6) with a significant F-value and p-value also suggests strong model significance.
4. **Roy's Greatest Root**: Like Hotelling-Lawley trace, a high value (1884.6) with a significant F-value and p-value indicates the model's significance.

### Method Part

1. **Wilks' Lambda**: A value of 0.1802 with an F-value of 3.3896 and a p-value of 0.0534. This p-value is marginally above the typical alpha level of 0.05, suggesting that the differences in group means are not quite statistically significant at the 5% level.
2. **Pillai's Trace**: A value of 0.8468, F-value of 2.2031, and a p-value of 0.1301. This result further indicates that the group means are not significantly different, as the p-value is above 0.05.
3. **Hotelling-Lawley Trace**: A value of 4.4 with an F-value of 5.4 and a p-value of 0.0444. This p-value is below 0.05, indicating significant differences in the group means.
4. **Roy's Greatest Root**: A value of 4.3656, with an F-value of 13.0969 and a p-value of 0.0065. This result suggests significant differences in the group means, as indicated by this low p-value.

### Overall Interpretation

- The significant intercept part indicates that the overall model is significant.
- For the `Method` part, different test statistics provide somewhat conflicting results. Wilks' Lambda and Pillai's Trace suggest that the means of different methods are not significantly different, while Hotelling-Lawley Trace and Roy's Greatest Root suggest significant differences.
- Such discrepancies can occur due to the sensitivity of each test to different assumptions and data characteristics. In practice, when results conflict, it's often advisable to further investigate the data, potentially considering other forms of analysis or looking into specific pairwise comparisons for more insights.

---